frontend/src/utils/wildcard.js

```diff
@@ -0,0 +1,71 @@
+//
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+'use strict'
+
+import startsWith from 'lodash/startsWith'
+import endsWith from 'lodash/endsWith'
+import map from 'lodash/map'
+import trim from 'lodash/trim'
+import filter from 'lodash/filter'
+import head from 'lodash/head'
+
+export function wildcardObjectsFromStrings (wildcardStrings) {
+  return map(wildcardStrings, item => {
+    let startsWithWildcard = false
+    let endsWithWildcard = false
+    let customWildcard = false
+    const value = trim(item, '*')
+    let pattern = value
+
+    if (item === '*') {
+      customWildcard = true
+      pattern = '.*'
+    } else if (startsWith(item, '*')) {
+      startsWithWildcard = true
+      pattern = '.*' + pattern
+    } else if (endsWith(item, '*')) {
+      endsWithWildcard = true
+      pattern = pattern + '.*'
+    }
+
+    return {
+      value,
+      originalValue: item,
+      startsWithWildcard,
+      endsWithWildcard,
+      customWildcard,
+      isWildcard: startsWithWildcard || endsWithWildcard || customWildcard,
+      regex: new RegExp('^' + pattern + '$')
+    }
+  })
+}
+
+export function bestMatchForString (wildCardObjects, wildCardString) {
+  const matches = filter(wildCardObjects, (item) => {
```

Comment:
We should first sort the pattern and then find the first match and return it.

Reply:
I did not change this method for now. If you have a proposal please go ahead.